apps/src/dropletUtils.js

```diff
@@ -404,7 +417,15 @@ exports.generateAceApiCompleter = function (functionFilter, dropletConfig) {
         callback(null, []);
         return;
       }
-      callback(null, apis);
+      var line = session.getLine(pos.row);
+      var identifier = acUtil.retrievePrecedingIdentifier(line, pos.column);
+      if (pos.column > (identifier.length + 1) &&
+          line[pos.column - identifier.length - 1] === '.') {
```

Comment:
Here's the [problem exploded using SlowmoJS](http://toolness.github.io/slowmo-js/?code=var%20apis%2C%20methodsAndProperties%3B%0Avar%20callback%20%3D%20function%20()%20%7B%7D%3B%0Avar%20ID_REGEX%20%3D%20%2F%5Ba-zA-Z_0-9%5C%24%5C-%5Cu00A2-%5CuFFFF%5D%2F%3B%0Avar%20retrievePrecedingIdentifier%20%3D%20function(text%2C%20pos%2C%20regex)%20%7B%0A%20%20%20%20regex%20%3D%20regex%20%7C%7C%20ID_REGEX%3B%0A%20%20%20%20var%20buf%20%3D%20%5B%5D%3B%0A%20%20%20%20for%20(var%20i%20%3D%20pos-1%3B%20i%20%3E%3D%200%3B%20i--)%20%7B%0A%20%20%20%20%20%20%20%20if%20(regex.test(text%5Bi%5D))%0A%20%20%20%20%20%20%20%20%20%20%20%20buf.push(text%5Bi%5D)%3B%0A%20%20%20%20%20%20%20%20else%0A%20%20%20%20%20%20%20%20%20%20%20%20break%3B%0A%20%20%20%20%7D%0A%20%20%20%20return%20buf.reverse().join(%22%22)%3B%0A%7D%3B%0A%0Avar%20pos%20%3D%20%7B%20column%3A%2012%20%7D%3B%0Avar%20line%20%3D%20%22var%20x%20%3D%20obj.%22%3B%20%2F%2Fsession.getLine(pos.row)%3B%0Avar%20identifier%20%3D%20retrievePrecedingIdentifier(line%2C%20pos.column)%3B%0Aif%20(pos.column%20%3E%20(identifier.length%20%2B%201)%20%26%26%0A%20%20%20%20line%5Bpos.column%20-%20identifier.length%20-%201%5D%20%3D%3D%3D%20%27.%27)%20%7B%0A%20%20%2F%2F%20Following%20a%20dot%2C%20we%20autocomplete%20from%20methodsAndProperties%3A%0A%20%20callback(null%2C%20methodsAndProperties)%3B%0A%7D%20else%20%7B%0A%20%20callback(null%2C%20apis)%3B%0A%7D&filterrange=30-30)